src/main/java/net/starlark/java/cmd/Main.java

```diff
@@ -71,6 +72,17 @@ private static String prompt() {
           return input.toString();
         }
         input.append(lineSeparator).append(line);
+
+        // Read lines until lines produce valid statements.
+        // TODO(adonovan): parse a compound statement, like the Python and
+        //   go.starlark.net REPLs. This requires a new grammar production, and
+        //   integration with the lexer so that it consumes new
+        //   lines only until the parse is complete.
+        StarlarkFile starlarkFile = StarlarkFile.parse(ParserInput.fromString(input.toString(), "<stdin>"));
+        if (starlarkFile.ok()) {
```

Comment:
My eagle-eyed coworker @brandjon pointed out during our internal code review that this will cause the REPL to stop reading a chunk prematurely if it reads a valid prefix at a line boundary, for example:
```
def f():
   pass
   pass
```
The principled fix is the one in the TODO comment above, but it's somewhat involved. An unprincipled fix would be to keep reading until a blank link if the parse returned a StarlarkFile whose last statement is an If/For/Def.